samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/ChatHistoryController.cs

```diff
@@ -164,18 +169,36 @@ public async Task<IActionResult> EditChatSessionAsync([FromBody] ChatSession cha
     {
         string chatId = chatParameters.Id;
 
-        ChatSession? chat = await this._chatSessionRepository.FindByIdAsync(chatId);
+        ChatSession? chat = await this._sessionRepository.FindByIdAsync(chatId);
         if (chat == null)
         {
             return this.NotFound($"Chat of id {chatId} not found.");
         }
 
         chat.Title = chatParameters.Title;
-        await this._chatSessionRepository.UpdateAsync(chat);
+        await this._sessionRepository.UpsertAsync(chat);
 
         return this.Ok(chat);
     }
 
+    /// <summary>
+    /// Service API to get a list of imported sources.
+    /// </summary>
+    [Authorize]
+    [Route("chatSession/{chatId:guid}/sources")]
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<IEnumerable<MemorySource>>> GetSourcesAsync(
+        [FromServices] IKernel kernel,
+        Guid chatId)
+    {
+        this._logger.LogInformation("Get imported sources of chat session {0}", chatId);
+
+        return this.Ok(await this._sourceRepository.FindByChatIdAsync(chatId.ToString()));
```

Comment:
Will this return 400 or 404? Should the API check if the chat exists in the source repository first?

Reply:
neither, the list will just be empty since we're returning an array here. we could check for chat existence, but I'm not sure it's needed since it will be an extra call and it seems like an edge case the user won't typically get into?

Reply:
I believe it's still valuable to return a 404. Sometimes developers will restart the service without refreshing the frontend while using volatile storage, which will result in sending a request with an outdated chat id.

Reply:
good point, thanks for bringing up this case. added 404 logic (and fixed it in other routes)